src/System.Text.Json/src/System/Text/Json/Serialization/ClassType.cs

```diff
@@ -7,20 +7,20 @@ namespace System.Text.Json
     /// <summary>
     /// Determines how a given class is treated when it is (de)serialized.
     /// </summary>
-    internal enum ClassType
+    internal enum ClassType : uint
```

Comment:
To me, it looks like a byte backed Flags enum would be better suited for what you are trying to do here.

Reply:
I thought a 32-bit value would be more performant, but there was no discernible difference for byte, so changing to byte.

Reply:
Presumably, we have structs that contain `ClassType` as fields/auto-props, so those struct sizes might get smaller now.